[nightly][copilot-shell] shell mode exit code 127 (command not found) lacks actionable error guidance

### Component

copilot-shell

### Bug Description

When a user runs a non-existent command in shell mode (e.g., `!nonexistentcmd123`), the CLI correctly returns exit code 127 and displays `bash: command not found`. However, the error output lacks actionable guidance — it only shows `Command exited with code 127.` without any keywords that help the user understand what went wrong or what to do next.

The test oracle (`experience-oracles.ts` → `ACTIONABLE_ERROR_PATTERNS`) requires error output to contain actionable keywords such as `error`/`failed`/`错误` (category 1) and `建议`/`请`/`try`/`check`/`检查`/`install` (category 2). The current output `Command exited with code 127.` matches neither, causing F02-05 test case to fail.

Root cause: `shellCommandProcessor.ts` handles all non-zero exit codes with a generic `Command exited with code ${result.exitCode}.` message, with no special handling for exit code 127 (command not found) to provide actionable guidance.

This issue was discovered by AgenticOS Nightly automated testing on 2026-07-03 (nightly-20260703-185424, anolisa main f13342ed) and confirmed as a product bug via triage.

### Steps to Reproduce

```shell
# Build and install copilot-shell RPM
cd /root/anolisa && bash scripts/rpm-build.sh copilot-shell
rpm -ivh --force scripts/rpmbuild/RPMS/x86_64/copilot-shell-*.rpm

# Run the failing test
cd /root/agentic-os-tests/cosh-tests
CLI_PATH=/usr/bin/cosh npx vitest run testcase/f02-shell-mode.test.ts -t 'F02-05'
# → FAIL: assertActionableError expects output to match ACTIONABLE_ERROR_PATTERNS
```

Manual reproduction:
```shell
cosh
# In the REPL, type:
!nonexistentcmd123
# Output: "Command exited with code 127.\nbash: nonexistentcmd123: command not found"
# No actionable keywords present → user gets no guidance
```

### Expected Behavior

When a command returns exit code 127 (command not found), the CLI should prepend an actionable error message containing keywords like "Error" and "Please check" / "try installing" so the user understands the failure and knows what to do next.

### Suggested Fix

```diff
--- a/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.ts
+++ b/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.ts
@@ -281,7 +281,20 @@ export const useShellCommandProcessor = (
                 finalOutput = `Command terminated by signal: ${result.signal}.\n${finalOutput}`;
               } else if (result.exitCode !== 0) {
                 finalStatus = ToolCallStatus.Error;
-                finalOutput = `Command exited with code ${result.exitCode}.\n${finalOutput}`;
+                // Exit code 127 typically means "command not found". Provide
+                // actionable guidance so the user understands the error and
+                // knows what to do next (check spelling, install the tool, etc.).
+                if (result.exitCode === 127) {
+                  finalOutput =
+                    `Error: command not found (exit code 127).\n` +
+                    `Please check the command name and spelling, or use ` +
+                    `'which <command>' to verify the command is installed.\n` +
+                    `If the command is not installed, try installing it or ` +
+                    `adding it to your PATH.\n\n` +
+                    finalOutput;
+                } else {
+                  finalOutput = `Command exited with code ${result.exitCode}.\n${finalOutput}`;
+                }
               }
```

Verified on ECS (47.239.51.100): Applied patch → `bash scripts/rpm-build.sh copilot-shell` → exit 0, RPM built successfully. Installed new RPM → ran `CLI_PATH=/usr/bin/cosh npx vitest run testcase/f02-shell-mode.test.ts` → all 10 F02 tests passed (including F02-05).

### Environment

- ALinux 4 ECS (cn-hongkong, 47.239.51.100)
- anolisa main: f13342edba8c80b2dc4768780cbbe21346b7bdcc
- copilot-shell 2.6.0

### Relevant Log Output

```
FAIL  testcase/f02-shell-mode.test.ts > F02 - ! Shell 模式 > F02-05: 执行不存在的命令
AssertionError: Output does not contain actionable error patterns
Expected output to match at least one pattern from ACTIONABLE_ERROR_PATTERNS
```

### Additional Context

Discovered and fixed by AgenticOS Nightly ([nightly] prefix). Fix verified green — all 10 F02 shell mode test cases pass.

--- a/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.test.ts
+++ b/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.test.ts
@@ -188,7 +188,7 @@ describe('useShellCommandProcessor', () => {
     expect(setShellInputFocusedMock).toHaveBeenCalledWith(false);
   });
 
-  it('should handle command failure and display error status', async () => {
+  it('should show actionable guidance for command not found', async () => {
     const { result } = renderProcessorHook();
 
     act(() => {
@@ -209,10 +209,40 @@ describe('useShellCommandProcessor', () => {
     const finalHistoryItem = addItemToHistoryMock.mock.calls[1][0];
     expect(finalHistoryItem.tools[0].status).toBe(ToolCallStatus.Error);
     expect(finalHistoryItem.tools[0].resultDisplay).toContain(
-      'Command exited with code 127',
+      'Error: command not found (exit code 127)',
+    );
+    expect(finalHistoryItem.tools[0].resultDisplay).toContain('Please check');
+    expect(finalHistoryItem.tools[0].resultDisplay).toContain(
+      'try installing it',
     );
     expect(finalHistoryItem.tools[0].resultDisplay).toContain('not found');
     expect(setShellInputFocusedMock).toHaveBeenCalledWith(false);
+  });
+
+  it('should preserve generic guidance for other command failures', async () => {
+    const { result } = renderProcessorHook();
+
+    act(() => {
+      result.current.handleShellCommand(
+        'bad-cmd',
+        new AbortController().signal,
+      );
+    });
+    const execPromise = onExecMock.mock.calls[0][0];
+
+    act(() => {
+      resolveExecutionPromise(
+        createMockServiceResult({ exitCode: 2, output: 'usage error' }),
+      );
+    });
+    await act(async () => await execPromise);
+
+    const finalHistoryItem = addItemToHistoryMock.mock.calls[1][0];
+    expect(finalHistoryItem.tools[0].status).toBe(ToolCallStatus.Error);
+    expect(finalHistoryItem.tools[0].resultDisplay).toContain(
+      'Command exited with code 2',
+    );
+    expect(finalHistoryItem.tools[0].resultDisplay).toContain('usage error');
   });
 
   describe('UI Streaming and Throttling', () => {

--- a/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.ts
+++ b/src/copilot-shell/packages/cli/src/ui/hooks/shellCommandProcessor.ts
@@ -281,7 +281,20 @@ export const useShellCommandProcessor = (
                 finalOutput = `Command terminated by signal: ${result.signal}.\n${finalOutput}`;
               } else if (result.exitCode !== 0) {
                 finalStatus = ToolCallStatus.Error;
-                finalOutput = `Command exited with code ${result.exitCode}.\n${finalOutput}`;
+                // Exit code 127 typically means "command not found". Provide
+                // actionable guidance so the user understands the error and
+                // knows what to do next (check spelling, install the tool, etc.).
+                if (result.exitCode === 127) {
+                  finalOutput =
+                    `Error: command not found (exit code 127).\n` +
+                    `Please check the command name and spelling, or use ` +
+                    `'which <command>' to verify the command is installed.\n` +
+                    `If the command is not installed, try installing it or ` +
+                    `adding it to your PATH.\n\n` +
+                    finalOutput;
+                } else {
+                  finalOutput = `Command exited with code ${result.exitCode}.\n${finalOutput}`;
+                }
               }
 
               if (pwdFilePath && fs.existsSync(pwdFilePath)) {

